main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintSolver.scala

```diff
@@ -480,13 +476,12 @@ object ConstraintSolver {
   }
 
   /**
-    * Gets an error from the list of unresolved constraints.
+    * Returns the errors from the list of unresolved constraints.
     */
-  private def getFirstError(deferred: List[TypeConstraint], renv: RigidityEnv)(implicit flix: Flix): Option[TypeError] = deferred match {
-    case Nil => None
-    case TypeConstraint.Equality(tpe1, tpe2, prov) :: _ => Some(toTypeError(UnificationError.MismatchedTypes(tpe1, tpe2), prov))
-    case TypeConstraint.Trait(sym, tpe, loc) :: _ => Some(mkMissingInstance(sym, tpe, renv, loc))
-    case TypeConstraint.Purification(_, _, _, _, nested) :: _ => getFirstError(nested, renv)
+  private def getErrorsFromTypeConstraints(deferred: List[TypeConstraint], renv: RigidityEnv)(implicit flix: Flix): List[TypeError] = deferred.flatMap {
```

Comment:
Hey! This isn't a refactor! :O 

Let's make this change separately. There might be unexpected consequences (like having 999999 errors)

Reply:
Yeah, that's also why I suggested making this change in a separate PR

Reply:
But just wanted to show that it was possible now